### ISSUE TITLE
Cross-Attention Maps for Text-guided Image-to-Image stable diffusion

### ISSUE THREAD

**Issue by nityanandmathur** [vpulab/ovam#1]
Hi! I used your work and it works amazingly great for T2I models.

Can this be extended to find out cross attention for Text-guided Image-to-Image(StableDiffusionImg2ImgPipeline) pipeline as well?

**Comment by pablomm** [comment]
Hi @nityanandmathur! Thank you for your interest. Yes, the method also works for `StableDiffusionImg2ImgPipeline`. 

Here you have an example on how to use it with initialization of images. A colab example with the full code is here https://colab.research.google.com/github/vpulab/ovam/blob/main/examples/ovam_img2img_colab.ipynb.

```python
device = "cuda"
model_id_or_path = "runwayml/stable-diffusion-v1-5"
pipe = StableDiffusionImg2ImgPipeline.from_pretrained(model_id_or_path, torch_dtype=torch.float16)
pipe = pipe.to(device)

url = "https://raw.githubusercontent.com/CompVis/stable-diffusion/main/assets/stable-samples/img2img/sketch-mountains-input.jpg"

response = requests.get(url)
init_image = Image.open(BytesIO(response.content)).convert("RGB")
init_image = init_image.resize((512, 512))

prompt = "A fantasy landscape, trending on artstation"

# Generate image storing the internal attentions
set_seed(1)
with StableDiffusionHooker(pipe) as hooker:
  image = pipe(prompt=prompt, image=init_image, strength=0.75, guidance_scale=7.5).images[0]

# Evaluator of attention
ovam_evaluator = hooker.get_ovam_callable(expand_size=(512, 512))

# Generate attetions of the string "A castle"
attribution_prompt = "A castle"
with torch.no_grad():
    attention_maps = ovam_evaluator(attribution_prompt)
    attention_maps = attention_maps[0].cpu().numpy() # (4, 512, 512) <SoT>A castle<EoT>
```

![example_img2img](https://github.com/vpulab/ovam/assets/16774925/457e5325-0675-4bae-ab41-8035578b7952)

During our initial experiments, we found that the greater the weight placed on the initialization image, the less the generation process relies on cross-attentions. Consequently, this can result in less accurate masks.

Note: While the method is applicable to images of any kind, the current implementation is limited to square images. If you wish to use it with non-square images, you'll need to modify the [`_compute_attentions`](https://github.com/vpulab/ovam/blob/d59d1162d7156949d5419e00a89629bba424a73a/ovam/stable_diffusion/daam_block.py#L51.) method.

**Comment by nityanandmathur** [comment]
Thank you very much for your prompt, elaborate and illustrative response! I am able to use this in my pipeline.
